src/test/java/com/powsybl/sld/server/SingleLineDiagramTest.java

```diff
@@ -331,4 +360,245 @@ public static Network createNetwork() {
 
         return network;
     }
+
+    @Test
+    public void testPosisionDiagramLabelProvider() throws FileNotFoundException {
+        var testNetwork = createNetworkWithOneInjection();
+        var layoutParameters = new LayoutParameters();
+        var componentLibrary = new ConvergenceComponentLibrary();
+        var graphBuilder = new NetworkGraphBuilder(testNetwork);
+        VoltageLevelGraph g = graphBuilder.buildVoltageLevelGraph("vl1");
+        PositionDiagramLabelProvider labelProvider = new PositionDiagramLabelProvider(testNetwork, componentLibrary, layoutParameters, "vl1");
+        PositionDiagramLabelProvider labelProvider2 = new PositionDiagramLabelProvider(testNetwork, componentLibrary, layoutParameters, "vl2");
+
+        List<FeederInfo> feederInfos1 = labelProvider.getFeederInfos((FeederNode) g.getNode("loadA"));
+        assertEquals(2, feederInfos1.size());
+        assertEquals(ARROW_ACTIVE, feederInfos1.get(0).getComponentType());
+        assertEquals(ARROW_REACTIVE, feederInfos1.get(1).getComponentType());
+        assertTrue(feederInfos1.get(0).getRightLabel().isPresent());
+        assertTrue(feederInfos1.get(1).getRightLabel().isPresent());
+        assertFalse(feederInfos1.get(0).getLeftLabel().isPresent());
+        assertFalse(feederInfos1.get(1).getLeftLabel().isPresent());
+        // test if position label successfully added to svg
+        DiagramStyleProvider diagramStyleProvider = new NominalVoltageDiagramStyleProvider(testNetwork);
+        SingleLineDiagram.draw(testNetwork, "vl1", Path.of("/tmp/test.svg"), layoutParameters, componentLibrary, labelProvider, diagramStyleProvider, "");
+        assertTrue(strExistInSvgFile("/tmp/test.svg", "loadA pos [0]"));
+        assertTrue(strExistInSvgFile("/tmp/test.svg", "trf1 pos [1]"));
+        assertTrue(strExistInSvgFile("/tmp/test.svg", "trf73 pos [3]"));
+        SingleLineDiagram.draw(testNetwork, "vl3", Path.of("/tmp/test2.svg"), layoutParameters, componentLibrary, labelProvider2, diagramStyleProvider, "");
+        assertTrue(strExistInSvgFile("/tmp/test2.svg", "trf71 pos [4]"));
+    }
+
+    private boolean strExistInSvgFile(String fileName, String searchStr) throws FileNotFoundException {
+        Scanner scan = new Scanner(new File(fileName));
+        while (scan.hasNext()) {
+            String line = scan.nextLine();
+            if (line.contains(searchStr)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public Network createNetworkWithOneInjection() {
```

Comment:
To replace with already configured FourSubstationsNodeBreakerWithExtensionsFactory when migrating to next powsybl release to avoid all this boiler plate code